dask_ml/datasets.py

```diff
@@ -451,8 +452,13 @@ def make_classification_df(
             [
                 X_df,
                 dd.from_array(
-                    np.array([random_date(*dates)] * len(X_df)),
-                    chunksize=chunks,
+                    np.array(
+                        [
+                            random_date(*dates, random_state + i)
```

Comment:
What happens when `random_state` isn't an integer?  Scikit-learn allows for `random_state` to be an integer, `None` or an instance of `np.random.RandomState` ([source][1]).

[1]:https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_random_state.html

Reply:
Okay, I will have to raise a ValueError exception there, on it.

Reply:
Why not use this code?

``` python
rng = check_random_state(random_state)
dates = [random_date(*dates, rng) for i in range(len(X_df))]
...
```

Reply:
The code above will produce the same random number since the seed(`rng`) remains the same in subsequent calls.

Reply:
My main point: I think `np.random.RandomState` and `None` should be acceptable types for `random_state`. I'm fine expanding the for-loop, though I don't think that needs to happen:

<details>

``` python
[ins] In [193]: def random_dates(random_state):
           ...:     return random_state.randint(100)
           ...:

[ins] In [194]: rng = np.random.RandomState(42)

[ins] In [196]: [random_dates(rng) for _ in range(20)]
Out[196]: [51, 92, 14, 71, 60, 20, 82, 86, 74, 74, 87, 99, 23, 2, 21, 52, 1, 87, 29, 37]
```


</details>

Reply:
I can maybe just check if random_state is one of the accepted values like this and accordingly proceed - 
`if random_state is not None or not isinstance(random_state, np.random.RandomState) or not isinstance(random_state,int):
    print("random_state is not to be accepted")`

Reply:
That runs counter to the use of `random_state` in Scikit-learn. `random_date` is public, so it should accept all types of `random_state` that Scikit-learn accepts.

If `random_date` were a private function, I wouldn't really care

Reply:
random_state : int, RandomState instance or None, optional (default=None), these are values accepted by Scikit-Learn's accepted values. I think I can check if the random_state is in neither of the accepted values(Scitkit and Numpy) and set is as the default None.

Reply:
Scikit-learn's `check_random_state` function will likely be useful: https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_random_state.html

It takes those values and produces the correct random seed generator.

Reply:
@stsievert the accepted types of `random_state` in Scikit and Numpy appear to be the same. 
Refer this: Scikit's version also accepts Numpy's accepted values. Refer this: https://scikit-learn.org/dev/glossary.html#term-random_state